inlämning vecka 2

Wuplx opened the pull request:
delat upp koden och lagt till tillståndshantering

Wuplx committed:
inlämning vecka 2

delat upp koden och lagt till tillståndshantering

Heijdenberg reviewed (commented):
Provider saknas I pubspec.yaml.

När man gör ett val i dropdown menyn så filtreras inte listan. Detta beror på att listen inte byggs om. 

Ett sätt att lösa det på är att i ”onChanged” i ”DropdownButton” så kan du ta bort ”setState” och lägga till ”notifyListeners”:
onChanged: (newValue) {
       context.read<MyState>().filter = newValue!;
       context.read<MyState>().notifyListeners();
}

I listviewbuilder.dart så kan du flytta ”context.read<MyState>().todolist.removeAt(index);” till en method, det fungerar som det är men flyttar du den så kan man använda den på fler ställen och man kan ta bort ”setState” och koden blir i mitt tycke mer läsbar. Du kan också låta ”context.read<MyState>().todolist.removeAt(index);” ligga kvar och ta bort ”setState” sedan lägga till  ”context.read<MyState>().notifyListeners();”.

I addview.dart så kan du ta bort ”setState”, för vad jag kan se så behövs den inte längre för att ”notifyListeners()” körs i ”addToList”.
Det samma gäller även ”Checkbox” där du kan ersätta ”setState” med ” context.read<MyState>().notifyListeners();”.

Om du väljer att göra dessa ändringar så att du inte längre behöver ”setState” så kan ”MyHome”, ” ListViewBuilder”, och ”AddView” vara ”StatelessWidget” i stället för en ”StatefulWidget” nu när du använder provider för att hantera State.

Inget måste för denna kurs men ”MyState”, ” MyHome” och ”TodoItem” skulle kunna flyttas ut till egna filer för att göra koden mer lätt läst.

Wuplx committed:
Delat upp koden mer och gjort ändringar enligt feedback

Delat upp koden i fler filer, ändrat en del klasser från statefull till stateless. Filtret fungerar som det ska igen. Pubspec är uppdaterad.

Wuplx commented:
Fixat så filtret fungerar igen och gjorde de flesta ändringar du rekommenderade!
